### DISCUSSION TITLE
Compile csproj /OC Module and reference DLL

### DISCUSSION THREAD

**Discussion by SzymonSel** [OrchardCMS/OrchardCore#15136] [General]
I expected that upon building a OrchardCore module/theme, I would simply need to reference the DLL in my startup project instead of the csproj, and all should work fine. Although the dll is copied into the outpub `bin` folder, the module/theme/features aren't available in OrchardCore. What am I missing?

**Comment by SzymonSel** [comment]
I went on and created a local nuget source, and consumed nuget packages instead of raw DLL. Still curious though...

**Reply by Piedone** [reply]
I'd expect DLLs to also work. My only guess is that perhaps you built them in some way differently when packaging them up? Perhaps the target framework?

**Reply by SzymonSel** [reply]
I double, triple checked that the target frameworks align, since this was my initial thought as I was building this to be used in docker.

**Comment by ns8482e** [comment]
You need to either reference the module's nuget or reference module's project to main app to work.

**Reply by SzymonSel** [reply]
Yah, that kinda blows in my scenario.
